### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Keanu-Ellwood-DVT**
- Remove unused methods in services.
- Update and add unit tests for components and services.

### CODE REVIEW

File: src/app/admin/new-book/new-book.component.spec.ts [hunk 1]
```diff
@@ -127,24 +185,106 @@ describe('NewBookComponent', () => {
   });
 
   it('should call the uploadPicture method', async () => {
-    spyOn(component, 'uploadPicture');
-    el = fixture.debugElement.query(By.css('input')).nativeElement;
+    const spy = spyOn(component, 'uploadPicture').and.callThrough();
+    el = fixture.debugElement.query(By.css('#customFile')).nativeElement;
     el.click();
```
Comment (strizzwald) [inline src/app/admin/new-book/new-book.component.spec.ts:190]: You should prevent this from open up the user's file explorer while executing these tests.

File: src/app/book-info/book-info.component.spec.ts [hunk 2]
```diff
@@ -26,29 +76,39 @@ describe('BookInfoComponent', () => {
...
     component = fixture.componentInstance;
     fixture.detectChanges();
-    modalService = TestBed.inject(NgbModal);
-    bookService = TestBed.inject(BookService);
+    spyModalService = TestBed.inject(NgbModal);
+    spyBookService = TestBed.inject(BookService);
+    spyAuthorService = TestBed.inject(AuthorService);
   });
 
   it('should create', () => {
```
Comment (strizzwald) [inline src/app/book-info/book-info.component.spec.ts:84]: This test already exists below.

File: src/app/search-results-list/search-results-list.component.spec.ts [hunk 3]
```diff
@@ -39,4 +55,103 @@ describe('SearchResultsListComponent', () => {
...
   });
+
+  it('should set tags', () => {
+    fixture.detectChanges();
+
+    expect(component.tags).toEqual([{} as Tag, ]);
+  });
+
+  it('should invoke searchBooks()', async () => {
+    // let response: Book[] = [];
```
Comment (strizzwald) [inline src/app/search-results-list/search-results-list.component.spec.ts:66]: Please remove the commented out code.

File: src/app/search-results-list/search-results-list.component.spec.ts [hunk 4]
```diff
@@ -39,4 +55,103 @@ describe('SearchResultsListComponent', () => {
...
+    });
+    it('should be invoked when checkbox clicked', async () => {
+      const tagCheck = fixture.debugElement.query(By.css('#Tag')).nativeElement;
+      tagCheck.disabled = false;
+      tagCheck.checked = true;
+      const spy = spyOn(component, 'onCheckboxChange').and.callThrough();
+      tagCheck.click();
+      expect(spy).toHaveBeenCalled();
+    });
+    // it('should remove tag from array when unchecked', async () => {
```
Comment (strizzwald) [inline src/app/search-results-list/search-results-list.component.spec.ts:95]: Please remove the commented out code.

File: src/app/search-results-list/search-results-list.component.spec.ts [hunk 5]
```diff
@@ -39,4 +55,103 @@ describe('SearchResultsListComponent', () => {
...
+    expect(component.tags).toEqual([{} as Tag, ]);
+  });
+
+  it('should invoke searchBooks()', async () => {
+    // let response: Book[] = [];
+    // let searchBookSpy = jasmine.createSpy('searchBookSpy');
+    spyOn(component, 'searchBooks').and.callThrough();
+
+    fixture.detectChanges();
+    // spyBookService.refreshRequired$.next();
```
Comment (strizzwald) [inline src/app/search-results-list/search-results-list.component.spec.ts:71]: Please remove the commented out code.

File: src/app/services/author.service.ts [hunk 6]
```diff
@@ -34,6 +34,6 @@ export class AuthorService {
           this.refreshRequired$.next();
         })
       )
-      .subscribe(data => { console.log('PUT request was successful. PUT: ', data); }, error => { console.log('Error', error); });
+      .subscribe();
```
Comment (strizzwald) [inline src/app/services/author.service.ts:37]: Prefer to always return the observable and let the user of the service call subscribe.

File: debug.log [hunk 7]
```diff
@@ -2,3 +2,63 @@
 [0507/100846.540:ERROR:process_info.cc(556)] ReadMemoryInfo failed
```
[the remark this answers is not in the document]
Comment (Keanu-Ellwood-DVT) [reply debug.log]: I removed it. Didn't notice that I added it

File: src/app/admin/admin.component.ts [hunk 8]
```diff
@@ -23,7 +23,6 @@ export class AdminComponent implements OnInit, OnDestroy {
   book: Book;
   newBook: Book;
```
[the remark this answers is not in the document]
Comment (Keanu-Ellwood-DVT) [reply src/app/admin/admin.component.ts:24]: This is an artifact from the original component. Variables, imports and http calls have all been removed now.

File: src/app/book-info/book-info.component.spec.ts (hunk 2 above)
Comment (Keanu-Ellwood-DVT) [reply src/app/book-info/book-info.component.spec.ts]: Removed duplicate test

File: src/app/admin/new-author/new-author.component.spec.ts [hunk 9]
```diff
@@ -38,9 +66,30 @@ describe('NewAuthorComponent', () => {
     expect(component).toBeTruthy();
   });
 
+  it('should call getAuthor and set newAuthor to return value', () => {
+    component.state = 'Update';
+    component.currentAuth = mockAuthor;
+    spyOn(spyAuthorService, 'getAuthor').and.returnValue(of(mockAuthor));
+    component.ngOnInit();
+    component.newAuthor = component.currentAuth;
```
[the remark this answers is not in the document]
Comment (Keanu-Ellwood-DVT) [reply src/app/admin/new-author/new-author.component.spec.ts]: Fixed by initializing the newAuthor in the component. newAuthor is now set correctly by the component

File: src/app/admin/new-author/new-author.component.spec.ts [hunk 10]
```diff
@@ -38,9 +66,30 @@ describe('NewAuthorComponent', () => {
...
+    component.newAuthor = mockAuthor;
+    const spy = spyOn(spyAuthorService, 'putAuthor').and.callFake(() => {});
+
+    component.addAuthor();
+
+    expect(component.submitted).toBeTrue();
+    expect(spy).toHaveBeenCalled();
+  });
+
   it('should set submitted to true', async () => {
```
[the remark this answers is not in the document]
Comment (Keanu-Ellwood-DVT) [reply src/app/admin/new-author/new-author.component.spec.ts]: Updated this test + others that were directly invoking a method to rather invoke methods on a ui element event trigger

File: src/app/admin/new-book/new-book.component.spec.ts (hunk 1 above)
Comment (Keanu-Ellwood-DVT) [reply src/app/admin/new-book/new-book.component.spec.ts]: Fixed. Updated all occurrences of fixture.debugElement that was testing a method invoking via UI element event trigger

File: src/app/services/author.service.ts (hunk 6 above)
Comment (Keanu-Ellwood-DVT) [reply src/app/services/author.service.ts:37]: Fixed this. Previously I had logged info and errors about requests, now the service user calls .subscribe on all service methods as needed.

**Pull request merged**